src/storage/data_table.cpp

```diff
@@ -101,8 +105,7 @@ TupleSlot DataTable::Insert(transaction::TransactionContext *txn,
   DeltaRecord *undo = txn->UndoRecordForInsert(this, accessor_.GetBlockLayout(), result);
 
   // Populate undo record with the before image of presence column
-  // TODO(Tianyu): This is projection list id, not col id, fix eventually
-  undo->Delta()->SetNull(VERSION_VECTOR_COLUMN_ID);
+  undo->Delta()->SetNull(0);
```

Comment:
Are we not using the VERSION_VECTOR_COLUMN_ID def anymore?

Reply:
As commented, this is the 0th index in the projection list id, which we assume to have col_id of 1.

This is confusing as hell. We will need to standardize these magic constants and document these assumptions later in the clean up.